django_project/geosight/data/utils.py

```diff
@@ -174,3 +177,48 @@ def run_zonal_analysis(
         elif aggregation == 'avg':
             aggregate = np.average(data) if data.size else None
         return aggregate
+
+
+
+def create_thumbnail(
+        input_path: typing.Union[str, os.PathLike],
+        output_path: typing.Union[str, os.PathLike],
+        thumbnail_size = (600, 380), quality: int = 85
+):
+    """
+    Resize a large image to a thumbnail size with high quality, preserving aspect ratio.  # noqa
+
+    Args:
+        input_path (str): Path to the input image file (JPG or PNG).
+        output_path (str): Path to save the thumbnail image.
+        thumbnail_size (tuple): Desired size of the thumbnail (width, height).
+        quality (int): Quality for the output JPEG (1-100).
+    """
+    try:
+        # Open the image
+        img = Image.open(input_path)
+
+        # Ensure the image is in RGB mode for JPEG compatibility
+        if img.mode in ("RGBA", "P"):
+            img = img.convert("RGBA")
+        else:
+            img = img.convert("RGB")
+
+        output_format = "JPEG"
+        if input_path.endswith(".png"):
+            output_format = "PNG"
+
+        # Preserve aspect ratio and create thumbnail
+        img.thumbnail(thumbnail_size, Image.Resampling.LANCZOS)
+
+        if input_path == output_path:
+            temp_file = tempfile.NamedTemporaryFile(delete=False)
+            img.save(temp_file.name, format=output_format, quality=quality)
+
+            # Replace the original file
+            shutil.move(temp_file.name, output_path)
```

Comment:
Hi @zamuzakki 
Please don't replace the original one
Just create new file and frontend will fetch the thumbnail one
Example: https://staging-geosight.unitst.org/media/icons/senegal_thumbnail.png
Maybe we can do : https://staging-geosight.unitst.org/media/icons/thumbnail/senegal_thumbnail.png

In frontend just append the thumbnail to the url

Reply:
Also handle the Image component if the https://staging-geosight.unitst.org/media/icons/thumbnail/senegal_thumbnail.png is failed
Can create new component for Image that can handle if src is failed